File: src/leetcode/array/T283.rs

pub struct Solution {

}
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut j = 0;
        for i in 0..nums.len() {
            if nums[i] != 0 {
                nums[j] = nums[i];
                if i != j {
                    nums[i] = 0;
                }
                j += 1;
            }
        }
    }
}

File: src/main/java/org/example/array/T283.java
package org.example.array;

import java.util.Arrays;

/**
 * 移动零
 * @Datetime: 15/12/2023 下午4:15
 * @Author: koopkl
 * @Link: <a href="https://leetcode.cn/problems/move-zeroes/description/"></a>
 */
public class T283 {
    public void moveZeroes(int[] nums) {
        int index = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] !=  0) {
                int temp = nums[i];
                nums[i] = nums[index];
                nums[index] = temp;
                index += 1;
            }
        }
    }
}
